django_project/api/osm_api_client.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+
+import time
+
+from osmapi import (
+    OsmApi, OsmApiError, ApiError,
+    ResponseEmptyApiError, ElementDeletedApiError)
+from requests_oauthlib import OAuth1
+
+
+class OAuthTokenMissingError(OsmApiError):
+    """
+    Error when oauth token is missing for an authenticated request
+    """
+    pass
+
+
+class OsmApiWrapper(OsmApi, object):
+
+    def __init__(
+            self,
+            client_key,
+            client_secret,
+            oauth_token,
+            oauth_token_secret,
+            api,
+            appid):
+        """OsmApi wrapper object constructor.
+        """
+        super(OsmApiWrapper, self).__init__(api=api, appid=appid)
+        self._client_key = client_key
+        self._client_secret = client_secret
+        self._oauth_token = oauth_token
+        self._oauth_token_secret = oauth_token_secret
+
+    def _http_request(self, method, path, auth, send, return_value=True):
+        """
+        Returns the response generated by an HTTP request.
+        `method` is a HTTP method to be executed
+        with the request data. For example: 'GET' or 'POST'.
+        `path` is the path to the requested resource relative to the
+        base API address stored in self._api. Should start with a
+        slash character to separate the URL.
+        `auth` is a boolean indicating whether authentication should
+        be preformed on this request.
+        `send` contains additional data that might be sent in a
+        request.
+        `return_value` indicates whether this request should return
+        any data or not.
+        If the oauth token is missing,
+        `OAuthTokenMissingError` is raised.
+        If the requested element has been deleted,
+        `OsmApi.ElementDeletedApiError` is raised.
+        If the response status code indicates an error,
+        `OsmApi.ApiError` is raised.
+        """
+        if self._debug:
```

Comment:
i see this function is called multiple times
i think you can create "Logger class, that will be print if debug=true"
so this should be just 1 line